src/npm/mod.rs

```diff
@@ -776,6 +774,28 @@ impl NpmPackageReq {
   }
 }
 
+impl Serialize for NpmPackageReq {
```

Comment:
This allows me to use this as a key in a hashmap that will be serialized and deserialized.